tscreen.go

```diff
@@ -1484,6 +1484,26 @@ func (t *tScreen) collectEventsFromInput(buf *bytes.Buffer, expire bool) []Event
 
 		partials := 0
 
+		if b[0] == '\x1b' && len(t.rawseq) > 0 {
+			strb := string(b)
+			completed := false
+			for _, r := range t.rawseq {
+				if strings.HasPrefix(strb, r) {
+					// a registered raw sequence matched the prefix
+					res = append(res, NewEventRaw(r))
+					t.escbuf.Reset()
+					for i := 0; i < len(r); i++ {
+						buf.ReadByte()
+					}
+					completed = true
+					break
+				}
+			}
+			if completed {
+				continue
+			}
+		}
```

Comment:
It seems it would be nice and natural to move this code to a separate function like `t.parseRegisteredRawSeq()`, similar to the below `t.parsePaste()`, `t.parseOSC52Paste()` etc.